docs/adrs/002-change-release-strategy.md

```diff
@@ -0,0 +1,78 @@
+---
+status: proposed
+date: 17-05-2024
+deciders: Kamil Kołodziej, Radosław Szuma, Jakub Pisarek
+informed: Fishjam Team, Cloud Team
+---
+# Change release strategy to eliminate downtimes
+
+## Context and Problem Statement
+
+In the current situation when we do a release we are shuting down the server and every room with that.
+This is an awful user experience. Another issue is the fact that any ongoing recordings are lost.
+
+## Decision Drivers
+
+* N o downtime while release happens
+* 0 lost recordings during release
+
+## Considered Options
+
+* Rolling update with eviction
+* Active room migration
+
+## Decision Outcome
+
+Chosen option: "Rolling update with eviction", because
+while second option sounds great, we acknowledge that solution is not trivial and we want to fix downtimes ASAP. First solution is a good starting point to remove existing problems and allow us to move forward.
+
+### Consequences
+
+Every new release will need to trigger fishjam process which will handle the shutdown. What that effectively means is that we are leaving the responsibility of triggering release to external orchestration tool (what is usually the case) and to monitor if the application exited correctly.
+
+External process will trigger fishjam shutdown process which is gonna mark application as one that no longer accepts the connections (new rooms etc.). Once every room is closed and all of the recordings are computed we are going to mark fishjam as ready to shutdown. This application instance should also be effectively removed from any external load balancing (k8s etc.) while it should still allow erlang clustering. Orchestrator process should query the fishjam till it's ready to shutdown and sends sigterm signal.
+
+This is applicable one by one for every instance in cluster, although it may take some time to release a new version and we may have 2 different versions on the cluster at the same time, we are accepting that tradeoff. We may consider `force` option to release a version no matter the state of fishjam (which may result in downtime).
+
+## Pros and Cons of the Options
+
+### Rolling update with eviction
+
+1. External orchestrator process triggers new release
+2. Starts new instance of fishjam with new version
+3. Fishjam shutdown process start
+4. Stop receiving traffic to that instance
+5. Wait till the last room is closed and all recordings are completed
+6. Finish fishjam shutdown process - ready for sigterm
+7. External orchestrator process kills fishjam instance
```

Comment:
**issue**: When the instance receives a `SIGTERM` - after cleaning up, it should quit by itself, not on an additional orchestrator request.